web/handlers/aliases.go

```diff
@@ -0,0 +1,153 @@
+package handlers
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"go.mindeco.de/http/render"
+	refs "go.mindeco.de/ssb-refs"
+
+	"github.com/ssb-ngi-pointer/go-ssb-room/aliases"
+	"github.com/ssb-ngi-pointer/go-ssb-room/roomdb"
+)
+
+// aliasHandler implements the public resolve endpoint for HTML and JSON requests.
+type aliasHandler struct {
+	r *render.Renderer
+
+	db roomdb.AliasService
+
+	muxrpcHostAndPort string
+	roomID            refs.FeedRef
+}
+
+func (a aliasHandler) resolve(rw http.ResponseWriter, req *http.Request) {
+	respEncoding := req.URL.Query().Get("encoding")
+
+	var ar aliasResponder
+	switch respEncoding {
+	case "json":
+		ar = newAliasJSONResponder(rw)
+	default:
+		ar = newAliasHTMLResponder(a.r, rw, req)
+	}
+
+	ar.UpdateRoomInfo(a.muxrpcHostAndPort, a.roomID)
+
+	name := mux.Vars(req)["alias"]
+	if name == "" && !aliases.IsValid(name) {
+		ar.SendError(fmt.Errorf("invalid alias"))
+		return
+	}
+
+	alias, err := a.db.Resolve(req.Context(), name)
+	if err != nil {
+		ar.SendError(err)
+		return
+	}
+
+	ar.SendConfirmation(alias)
+}
+
+// aliasResponder is supposed to handle different encoding types transparently.
+// It either sends the signed alias confirmation or an error.
+type aliasResponder interface {
+	SendConfirmation(roomdb.Alias)
+	SendError(error)
+
+	UpdateRoomInfo(hostAndPort string, roomID refs.FeedRef)
+}
+
+// aliasJSONResponse dictates the field names and format of the JSON response for the alias web endpoint
+type aliasJSONResponse struct {
+	Status    string `json:"status"`
+	Address   string `json:"address"`
+	RoomID    string `json:"roomId"`
+	UserID    string `json:"userId"`
+	Alias     string `json:"alias"`
+	Signature string `json:"signature"`
+}
+
+// handles JSON responses
+type aliasJSONResponder struct {
+	enc *json.Encoder
+
+	roomID        refs.FeedRef
+	multiservAddr string
+}
+
+func newAliasJSONResponder(rw http.ResponseWriter) aliasResponder {
+	return &aliasJSONResponder{
+		enc: json.NewEncoder(rw),
+	}
+}
+
+func (json *aliasJSONResponder) UpdateRoomInfo(hostAndPort string, roomID refs.FeedRef) {
+	json.roomID = roomID
+
+	roomPubKey := base64.StdEncoding.EncodeToString(roomID.PubKey())
+	json.multiservAddr = fmt.Sprintf("net:%s~shs:%s", hostAndPort, roomPubKey)
+}
+
+func (json aliasJSONResponder) SendConfirmation(alias roomdb.Alias) {
+	var resp = aliasJSONResponse{
+		Status:    "successfull",
```

Comment:
Is this status field relevant only internally (e.g. tests in go-ssb-room) or also externally? If externally, then it's important to fix this typo "successfull" to "successful". Should I also include `status`/`error` fields in the spec?

Reply:
```suggestion
		Status:    "successful",
```

Reply:
Yay, first commit suggestion I make

Reply:
This comes back to https://github.com/ssb-ngi-pointer/rooms2/issues/28

I find this useful if you want to check the schema of the returned json object. ie there is something to decide if the schema should be valid. And if it's not there also is the `error` field as a message to go with. Not just the 404 or 500 status code.